src/test/java/seedu/address/logic/commands/RedoCommandTest.java

```diff
@@ -0,0 +1,232 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.util.Comparator;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.CommandResult;
+import seedu.address.logic.commands.results.RedoCommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.item.Internship;
+import seedu.address.model.item.Person;
+import seedu.address.model.item.Project;
+import seedu.address.model.item.Resume;
+import seedu.address.testutil.InternshipBuilder;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.ResumeBuilder;
+import seedu.address.testutil.TypicalInternship;
+import seedu.address.testutil.TypicalResume;
+import seedu.address.testutil.TypicalResumeBook;
+
+
+public class RedoCommandTest {
+    private Model expectedModel = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+    private Model model = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+
+    @BeforeEach
+    public void setUp() {
+        expectedModel = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+        model = new ModelManager(TypicalResumeBook.TYPICAL_WITH_FILLED_RESUME, new UserPrefs());
+    }
+
+    @Test
+    public void execute_setUser_success() {
+        Person newUser = new PersonBuilder(model.getUser()).withName("Michael Ng").build();
+        model.setUser(newUser);
+        model.commitResumeBook();
+        model.undoResumeBook();
+
+        expectedModel.setUser(newUser);
+        expectedModel.commitResumeBook();
+        expectedModel.undoResumeBook();
+        expectedModel.redoResumeBook();
```

Comment:
I have slight reservations about using `undo` to test `redo`, because we should test them in isolation. However, I don't see any other way. So I suppose this is okay for now

Reply:
We use the `undo` and `redo` function of `Model` so it should be alright.